gguf-py/gguf/gguf.py

```diff
@@ -501,13 +511,26 @@ def add_string(self, key: str, val: str):
         self.add_key(key)
         self.add_val(val, GGUFValueType.STRING)
 
-    def add_array(self, key: str, val: list):
-        if not isinstance(val, list):
-            raise ValueError("Value must be a list for array type")
+    def add_array(self, key: str, val: Sequence[Any]):
+        if not isinstance(val, collections_abc.Sequence):
```

Comment:
You can use `typing.Sequence` here. Seems to work in Python 3.6 and newer at least, I don't have any older versions on hand.

Reply:
Ahh, that's nicer. I have Python 3.11 so I can't test with older versions either.

Anyone have an objection to this approach or evidence it doesn't work on older versions?